packages/rocketchat-lib/server/functions/setUserAvatar.js

```diff
@@ -11,7 +11,7 @@ RocketChat.setUserAvatar = function(user, dataURI, contentType, service) {
 		let result = null;
 
 		try {
-			result = HTTP.get(dataURI, { npmRequestOptions: { encoding: 'binary' } });
+			result = HTTP.get(dataURI, { npmRequestOptions: { encoding: 'binary', rejectUnauthorized: false } });
```

Comment:
I don’t know that this really solves the issue.  It just makes it so invalid ssl doesn’t hold it up.  But what if a server instead did something else to cause major delay in response? 

Also, do we really want to ignore invalid ssl?  

I wonder if instead we could make window shorter that it will wait on a response?

Reply:
That’s basically what this is doing.  It’s explicitly saying not to use rejectUnauthorized.  It’s true by default if not specified.

I agree though if anything it should be admin option.  

But I stand by that there has to be a better way to solve the issue.  Since the issue isn’t necessarily about invalid ssl.  But more about the server hanging